pkgs/applications/misc/st/default.nix

```diff
@@ -1,33 +1,32 @@
-{ stdenv, fetchurl, pkgconfig, writeText, libX11, ncurses, libXext, libXft, fontconfig
-, conf ? null, patches ? []}:
+{ stdenv, fetchurl, pkgconfig, writeText, libX11, ncurses, libXext, libXft
+, fontconfig, conf ? null, patches ? []}:
 
 with stdenv.lib;
 
-stdenv.mkDerivation rec {
-  name = "st-0.6";
-  
+let patches' = if isNull patches then [] else patches;
```

Comment:
1. all-packages.nix passes in `patches` as `null` if the relevant config value is not set
2. `mkDerivation` is called with a recursive attribute set, so `patches` is bound to the `patches` attribute, not the `patches` argument, inside the set.